app/src/main/java/ch/epfl/sweng/erpa/activities/GameViewerActivity.java

```diff
@@ -1,188 +1,252 @@
-        if (gameId == null) {
-            Exception thrown = new IllegalArgumentException("Game Id not found");
-            Log.d(TAG, "GameViewerActivity: no game id passed with intent", thrown);
-            finish();
-            throw new IllegalArgumentException();
+        if (!modifyingSelfRequest) {
+            createPopup("Naughty naughty, you can't modify other people's requests!", this);
         } else {
-            return gameId;
+            createPopup("I know you're super excited about it!", this);
         }
     }
 
-    @OnClick(R.id.joinGameButton)
-    void joinGame() {
-        game = game.withPlayer(up.getUuid());
-        gs.saveGame(game);
-        updateGame();
+    @Override protected void onStop() {
+        super.onStop();
+        Stream.of(asyncFetchThreads.values()).forEach(v -> v.cancel(true));
     }
 
-    private void updateGame() {
-        getGameOrFinish(gs.getGame(getGameId()));
-        updateFields();
+    @OnClick(R.id.joinGameButton)
+    void joinGame() {
+        Intent joinGame = new Intent(this, JoinGameActivity.class);
+        joinGame.putExtra(JoinGameActivity.GAME_UUID_KEY, gameUuid);
+        startActivity(joinGame);
     }
 
-    @SuppressLint("SetTextI18n")
-    private void updateFields() {
-        Log.d(TAG, "onResume: Successfully fetched game");
-
-        title.setText(game.getName());
+    @UiThread
+    private void updateGameTVs(Game game) {
+        String gmUserUuid = game.getGmUserUuid();
+        asyncFetchThreads.put(gmUserUuid, asyncTaskService.run(
+            () -> ups.getUsernameFromUserUuid(gmUserUuid).map(Username::getUsername),
+            failIfNotFound(gmUserUuid, gm -> {
+                gmName.setVisibility(View.VISIBLE);
+                gmName.setText(gm);
+            })
+        ));
+
+        contentPanel.setVisibility(View.VISIBLE);
+        panelLoader.setVisibility(View.GONE);
+
+        title.setText(game.getTitle());
         description.setText(game.getDescription());
-        gmName.setText(game.getGmUuid());
         universe.setText(game.getUniverse());
         difficulty.setText(game.getDifficulty().toString());
-        type.setText(game.getOneshotOrCampaign().toString());
+        type.setText(game.getOneshotOrCampaign());
 
-        String numSessionsString = game.getNumberSessions().map(Object::toString).orElse("Unspecified");
+        String numSessionsString = game.getNumberOfSessions().map(Object::toString).orElse("Unspecified");
         numSessions.setText(numSessionsString);
 
         String gameLength = game.getSessionLengthInMinutes().map(Object::toString).orElse("Unspecified");
         sessionLength.setText(gameLength);
+    }
 
+    @UiThread
+    private void updateGameJoinRequests(List<PlayerJoinGameRequest> gameParticipantRequests, Game game) {
+        Log.d("GV FetchParticipants", "Updating game participants: " + gameParticipantRequests.toString());
```

Comment:
Don't forget to remove this

Reply:
Why? I think this log is usefull

Reply:
I asked you when I was reviewing and you said that I should put a reminder for you there. But yeah I don't have any real reason why to remove it